src/lib/oauth2-service.ts

```diff
@@ -314,15 +311,17 @@ export class OAuth2Service extends EventEmitter {
         this.#nonce[code] = queryNonce;
       }
       url.searchParams.set('code', code);
-      url.searchParams.set('scope', scope);
-      url.searchParams.set('state', state);
     } else {
       url.searchParams.set('error', 'unsupported_response_type');
       url.searchParams.set(
         'error_description',
         'The authorization server does not support obtaining an access token using this response_type.'
       );
-      url.searchParams.set('state', state);
+    }
+
+    if ('state' in req.query) {
```

Comment:
@jirutka Could you please add another test covering the case when state isn't passed in?

Reply:
Okay, added.